validation-test/stdlib/SetAlgebra.swift

```diff
@@ -0,0 +1,181 @@
+  }
+
+  public func intersection(_ other: DefaultedSetAlgebra) ->
+  DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.intersection(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func symmetricDifference(_ other: DefaultedSetAlgebra) ->
+  DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.symmetricDifference(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public mutating func insert(_ newMember: Element) -> (inserted: Bool,
+  memberAfterInsert: Element) {
+    return store.insert(newMember)
+  }
+
+  public mutating func remove(_ member: Element) -> Element? {
+    return store.remove(member)
+  }
+
+  public mutating func update(with newMember: Element) -> Element? {
+    return store.update(with: newMember)
+  }
+
+  public mutating func formUnion(_ other: DefaultedSetAlgebra) {
+    store.formUnion(other.store)
+  }
+
+  public mutating func formSymmetricDifference(_ other: DefaultedSetAlgebra) {
+    store.formSymmetricDifference(other.store)
+  }
+
+  public mutating func formIntersection(_ other: DefaultedSetAlgebra) {
+    store.formIntersection(other.store)
+  }
+}
+
+public func ==<Element>(lhs: DefaultedSetAlgebra<Element>,
+  rhs: DefaultedSetAlgebra<Element>) -> Bool {
+  return lhs.store == rhs.store
+}
+
+let SetAlgebraTests = TestSuite("SetAlgebra")
+
+SetAlgebraTests.test("contains") {
+  typealias P = DefaultedSetAlgebra<MinimalEquatableValue>
+  for test in findTests {
+    expectEqual(test.expected != nil,
+      P(test.sequence).contains(test.element))
+  }
+}
+
+SetAlgebraTests.test("union") {
+  for test in unionTests {
+    expectEqual(DefaultedSetAlgebra(test.expected),
+      DefaultedSetAlgebra(test.a).union(DefaultedSetAlgebra(test.b)))
+  }
+}
+
+SetAlgebraTests.test("intersection") {
+  for test in intersectionTests {
+     expectEqual(DefaultedSetAlgebra(test.expected),
+      DefaultedSetAlgebra(test.a).intersection(DefaultedSetAlgebra(test.b)))
+  }
+}
+
+SetAlgebraTests.test("symmetricDifference") {
+  for test in symmetricDifferenceTests {
+    let sequence =  DefaultedSetAlgebra(test.b)
+    expectEqual(DefaultedSetAlgebra(test.expected),
+      DefaultedSetAlgebra(test.a).symmetricDifference(sequence))
+  }
+}
+
+SetAlgebraTests.test("insert") {
+  for test in findTests {
+    var sequence = DefaultedSetAlgebra(test.sequence)
+    expectEqual((test.expected == nil, test.element),
+      sequence.insert(test.element))
+  }
+}
+
+SetAlgebraTests.test("subtracting") {
+  for test in subtractingTests {
+    expectEqual(DefaultedSetAlgebra(test.expected),
+      DefaultedSetAlgebra(test.a).subtracting(DefaultedSetAlgebra(test.b)))
+  }
+}
+
+SetAlgebraTests.test("isSubset") {
+  for test in findTests {
+    let sequence = DefaultedSetAlgebra(test.sequence)
+    expectEqual(test.expected != nil,
+      DefaultedSetAlgebra([test.element]).isSubset(of: sequence))
```

Comment:
This test is a great start, but please leave a FIXME to add more diverse tests where the base collection is not just a single-element one.